pkg/scheduler/api/podgroup_info/job_info.go

```diff
@@ -209,7 +199,14 @@ func (pgi *PodGroupInfo) setSubGroups(podGroup *enginev2alpha2.PodGroup) error {
 		return err
 	}
 	pgi.RootSubGroupSet = rootSubGroupSet
-	pgi.PodSets = rootSubGroupSet.GetPodSets()
+	podSets := rootSubGroupSet.GetPodSets()
+	if len(podSets) > 0 {
+		pgi.PodSets = podSets
+	} else {
+		for _, podSet := range pgi.PodSets {
+			rootSubGroupSet.AddPodSet(podSet)
```

Comment:
I dont understand how in the "SetSubGroup", which again happen directly after "NewPodGroup" the `pgi.PodSets` has some value

Reply:
OK Got it, when someone creates PodGroupInfo and not running "SetSubGroup" it will have the default SubGroup.
Happens in tests only.
Again its code that exists for tests only in prod structs